pychromecast/socket_client.py

```diff
@@ -231,8 +232,10 @@ def initialize_connection(self):
 
                 self._report_connection_status(
                     ConnectionStatus(CONNECTION_STATUS_FAILED, None))
-                self.logger.exception("Failed to connect, retrying in %fs",
-                                      self.retry_wait)
+                retry_log_fun("Failed to connect, retrying in %fs",
+                              self.retry_wait)
+                retry_log_fun = self.logger.debug
```

Comment:
The way I have it now, an error will always be thrown on the first failed
connection attempt. Every time pychromecast has to reconnect, this will be
reset.

I think this should be ok. In regular use, if pychromecast isn't working,
there will be some messages letting you know that it dropped a connection
and had trouble reconnecting. If someone were to have a lot of issues, they
should log info and debug messages to get more detail on what is happening.

Now that I'm thinking about this again, though. Maybe this should log to
info rather than debug after repeating failed retries. If you'd prefer, I
can also elevate it back to exception after x failed retries. I didn't do
that because of the reasons discussed in #60 with users turning their Cast
off when their tv is off.

On Saturday, January 9, 2016, Paulus Schoutsen <notifications@github.com>
wrote:

> In pychromecast/socket_client.py
> <https://github.com/balloob/pychromecast/pull/71#discussion_r49261442>:
>
> > @@ -231,8 +232,10 @@ def initialize_connection(self):
> >
> >                  self._report_connection_status(
> >                      ConnectionStatus(CONNECTION_STATUS_FAILED, None))
> > -                self.logger.exception("Failed to connect, retrying in %fs",
> > -                                      self.retry_wait)
> > +                retry_log_fun("Failed to connect, retrying in %fs",
> > +                              self.retry_wait)
> > +                retry_log_fun = self.logger.debug
>
> Could there be a moment you want to reset this one to exception?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/pychromecast/pull/71/files#r49261442>.
>


-- 
Thanks,

Ryan Kraus